scrypto/src/component/stubs.rs

```diff
@@ -239,15 +240,16 @@ extern_blueprint_internal! {
         ) -> Global<MultiResourcePool>;
     },
     {
-        fn contribute(&mut self, buckets: Vec<Bucket>) -> (Bucket, Vec<Bucket>);
+        fn contribute(&mut self, buckets: Vec<FungibleBucket>)
+            -> (FungibleBucket, Vec<FungibleBucket>);
         fn redeem(&mut self, bucket: Bucket) -> Vec<Bucket>;
-        fn protected_deposit(&mut self, bucket: Bucket);
+        fn protected_deposit(&mut self, bucket: FungibleBucket);
         fn protected_withdraw(
             &mut self,
             resource_address: ResourceAddress,
```

Comment:
Do we want to double down on the whole fungible and non-fungible bit and make this a `FungibleResourceManager`?

Reply:
Done. 
As agreed refactored other parts, where `ResourceAddress` appears.
Switched to `FungibleResourceManager`, `NonFungibleResourceManager` or `ResourceManager` depending on context.